src/types.ts

```diff
@@ -33,11 +33,14 @@ export type BlueProxy = BlueObject | BlueFunction;
 
 export type DistortionMap = WeakMap<RedProxyTarget, RedProxyTarget>;
 
+export type RedToBlueReferencesMap = WeakMap<RedProxy, RedProxyTarget | BlueProxy>;
```

Comment:
I'm supportive of this change... makes sense... but I believe the types are incorrect, they do not match the original implementation. In those maps we not only have proxies, we could have regular objects as well.

Reply:
Ah yes, I should have mentioned the steps I followed:
```ts
// Original implementation:

export type RedProxy = RedObject | RedFunction;
export type BlueProxy = BlueObject | BlueFunction;

export interface MembraneBroker {
    redMap: WeakMap<RedFunction | RedObject, RedProxyTarget | BlueProxy>;
    blueMap: WeakMap<BlueFunction | BlueObject, RedProxy | BlueProxyTarget>;
```
```ts
// Step 1: abstract WeakMap types from MembraneBroker interface

export type RedProxy = RedObject | RedFunction;
export type BlueProxy = BlueObject | BlueFunction;

export type RedToBlueReferencesMap = WeakMap<RedFunction | RedObject, RedProxyTarget | BlueProxy>;
export type BlueToRedReferencesMap = WeakMap<BlueFunction | BlueObject, RedProxy | BlueProxyTarget>;

export interface MembraneBroker {
    redMap: RedToBlueReferencesMap;
    blueMap: BlueToRedReferencesMap;
```
```ts
// Step 2: replace first unions with pre-existing RedProxy and BlueProxy types 
// and reordered second union types

export type RedProxy = RedObject | RedFunction;
export type BlueProxy = BlueObject | BlueFunction;

export type RedToBlueReferencesMap = WeakMap<RedProxy, RedProxyTarget | BlueProxy>;
export type BlueToRedReferencesMap = WeakMap<BlueProxy, BlueProxyTarget | RedProxy>;
```

Reply:
Maybe I am misusing `RedProxy` and `BlueProxy` here?